server/service/osquery_utils/queries.go

```diff
@@ -1214,29 +1214,38 @@ var SoftwareOverrideQueries = map[string]DetailQuery{
 	//     (having big queries can cause performance issues or be denylisted).
 	"macos_codesign": {
 		Query: `
-		SELECT a.path, c.team_identifier
+		SELECT a.path, c.team_identifier, c.cdhash_sha256
```

Comment:
Because this is on a custom table only present in fleet, do we have to do one of those `Discovery` queries prior to running this to "ensure the column is present"?

Reply:
Deferring to @lucasmrod on this, but _guessing_ that we'll still need a discovery query here to check for the column, otherwise we'll have cases where vitals break IIRC.

One thought here would be to revise the _existing_ discovery query to only return true if we have both the table and the new column, then make sure the newer fleetd gets released first. That saves us from having to discover 2x, and as long as folks are as up-to-date on fleetd as they are on Fleet server (which should be easy, particularly if we release fleetd first), they'll gain info rather than losing it. If folks care about codesign and see it drop, "make sure you're on current fleetd" is a straightforward remediation.

Reply:
Okay, [we're good](https://fleetdm.slack.com/archives/C062D0THVV1/p1745625487064529?thread_ts=1745625180.148559&cid=C062D0THVV1) to tweak the existing discovery query rather than trying to keep codesign backward-compat.

Reply:
Will need to swap the `discoveryTable` call for something fancier to make sure we don't query a column that doesn't exist (i.e. we should be backward-compatible server-side with current fleetd).

Reply:
To check if the table AND the column exists we can do something like this:
```
Discovery: generateSQLForAllExists(
  `SELECT COUNT(*) > 0 AS column_exists FROM pragma_table_info('codesign') WHERE name = 'cdhash_sha256'`,
  discoveryTable("codesign"),
),
```